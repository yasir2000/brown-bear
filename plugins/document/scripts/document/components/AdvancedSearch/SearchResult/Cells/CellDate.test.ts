/**
 * Copyright (c) Enalean, 2022 - present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import { shallowMount } from "@vue/test-utils";
import localVue from "../../../../helpers/local-vue";
import { createStoreMock } from "@tuleap/core/scripts/vue-components/store-wrapper-jest";
import type { ConfigurationState } from "../../../../store/configuration";
import CellDate from "./CellDate.vue";

describe("CellDate", () => {
    it("should display the update date of the item", () => {
        const wrapper = shallowMount(CellDate, {
            localVue,
            propsData: {
                date: "2021-10-06",
            },
            mocks: {
                $store: createStoreMock({
                    state: {
                        configuration: {
                            date_time_format: "Y-m-d H:i",
                            relative_dates_display: "relative_first-absolute_shown",
                            user_locale: "en_US",
                        } as unknown as ConfigurationState,
                    },
                }),
            },
            stubs: {
                "tlp-relative-date": true,
            },
        });

        expect(wrapper.element).toMatchSnapshot();
    });
});
