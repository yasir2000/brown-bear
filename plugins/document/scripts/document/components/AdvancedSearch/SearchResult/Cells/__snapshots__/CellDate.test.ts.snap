// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellDate should display the update date of the item 1`] = `
<td>
  <tlp-relative-date-stub
    absolute-date="2021-10-06 00:00"
    date="2021-10-06"
    locale="en_US"
    placement="top"
    preference="relative"
  >
    
        2021-10-06 00:00
    
  </tlp-relative-date-stub>
</td>
`;
